src/com/mendix/widget/ProgressCircle/components/ProgressCircle.ts

```diff
@@ -0,0 +1,148 @@
+    }
+
+    componentWillUnmount() {
+        this.progressCircle.destroy();
+    }
+
+    private createProgressCircle() {
+        this.progressCircle = new Circle(this.progressNode, {
+            duration: this.props.animate ? 800 : -1,
+            strokeWidth: 6,
+            trailWidth: 6
+        });
+        this.progressCircle.trail.className.baseVal = "widget-progress-circle-trail-path";
+        this.progressCircle.path.className.baseVal = "widget-progress-circle-path";
+    }
+
+    private setProgress(value: number, maximum: number) {
+        if (!this.progressCircle) {
+            this.createProgressCircle();
+        }
+
+        let progress = 0;
+        let progressText: string;
+        if (value === null || typeof value === "undefined") {
+            progressText = "--";
+        } else if (maximum <= 0) {
+            window.console.warn("The maximum value is 0. Progress is set to Invalid");
+            progressText = "Invalid";
+        } else {
+            progress = Math.round((value / maximum) * 100);
+            progressText = progress + "%";
+        }
+
+        let animateValue = progress / 100;
+        animateValue = animateValue > 1 ? 1 : animateValue < -1 ? -1 : animateValue;
+
+        this.progressCircle.setText(progressText);
+        this.progressCircle.animate(animateValue);
+    }
+
+    private checkConfig() {
+        let errorMessage: string[] = [];
+        if (this.props.progressOnClick.onClickType === "callMicroflow"
+            && !this.props.progressOnClick.microflowProps.microflow) {
+            errorMessage.push("'On click' call a microFlow is set " +
+                "and there is no 'Microflow' Selected in tab Events");
+        }
+        if (this.props.progressOnClick.onClickType === "showPage" && !this.props.progressOnClick.pageProps.page) {
+            errorMessage.push("'On click' Show a page is set and there is no 'Page' Selected in tab 'Events'");
+        }
+        if (errorMessage.length > 0) {
+            errorMessage.unshift("Error in configuration of the Progress circle widget");
+            window.mx.ui.error(errorMessage.join("\n"));
+        }
+    }
+
+    private handleOnClick(props: OnclickProps) {
+        if (props.onClickType === "callMicroflow" && props.microflowProps.microflow && props.microflowProps.guid) {
+            window.mx.ui.action(props.microflowProps.microflow, {
+                error: error =>
+                    window.mx.ui.error(
+                        `Error while executing microflow: ${props.microflowProps.microflow}: ${error.message}`
+                    ),
+                params: {
+                    applyto: "selection",
+                    guids: [ props.microflowProps.guid ]
+                }
+            });
+        } else if (props.onClickType === "showPage" && props.pageProps.page && props.pageProps.guid) {
+            let context = new window.mendix.lib.MxContext();
+            context.setTrackId(props.pageProps.guid);
```

Comment:
Pull request from Guy is pending. Changed the `package.json` link to point to that till the pull request is accepted.